### PR TITLE
fix(cbx-reader): fix two-page layout + navigation

### LINKED ISSUE #1029: CBX reader two-page mode messes spreads up/breaks paginated fit modes
### Discussed in https://github.com/orgs/grimmory-tools/discussions/1028

<div type='discussions-op-text'>

<sup>Originally posted by **zachyale** May  1, 2026</sup>
### What went wrong?

The CBX reader is effectively broken in paginated two page views
- Two page spreads can render in the wrong order
- They can show only one page when a spread should be shown
- They navigate inconsistently

Sometimes, they might show page 2 by itself, then page 3/4 together, then page 4 by itself, then page 5 with page 4 on the right instead of the left. It's bad.

On top of this, paginated fit modes also behave strangely in two page views:
- fit height stretches the spead to fill the width instead of preserving height/allowing horizontal scrolling
- actual size zooms correctly, but the scroll area is too small
- fit width could end up offset vertically
- images also have an annoying raise effect on hover

### How can we reproduce it?

1. Literally try to read a comic

### What Build of Grimmory are you on?

Stable

### Your setup

- Grimmory nightly

### Screenshots or error messages (optional)

_No response_

### Before submitting

- [X] I've searched existing discussions and this hasn't been reported yet
- [X] I've reviewed this report and have completed it as fully as I can</div>

### PR DESCRIPTION
**Pull request by zachyale**
## Description

Fixes CBX reader regressions in paginated two-page mode by unifying spread calculation for rendering, navigation, and alignment, and by tightening the paginated fit-mode layout rules.

**Linked Issue:** Fixes https://github.com/grimmory-tools/grimmory/issues/1029

## Changes

- unify CBX two-page spread handling across rendering, navigation, and alignment
- fix paginated `fit-width`, `fit-height`, and `actual-size` behavior in two-page view
- remove the CBX page hover zoom effect
- add focused spread utility tests

## Summary by CodeRabbit

* **Bug Fixes**
  * Two-page mode now computes and navigates spreads reliably, snapping to the correct spread and correctly handling single- vs two-page spreads during transitions.

* **Style**
  * Removed hover-zoom on page images.
  * Improved centering and sizing across fit modes (width/height/page/auto) and refined two-page layout behavior.

* **Refactor**
  * Default page-spread parity changed from odd to even.

* **Tests**
  * Added tests for spread computation and lookup behavior.

### CODE REVIEW

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss [hunk 1]
```diff
@@ -335,17 +332,43 @@
...
+            width: max-content;
+            height: auto;
+            justify-content: center;
+          }
+
           .page-image {
             width: auto;
-            height: 100%;
-            max-width: calc(50% - 0.5rem);
+            height: calc(100dvh - 1rem);
```
Comment (imnotjames) [inline frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss:367]: Where is the `1rem` coming from?

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss [hunk 2]
```diff
@@ -274,22 +274,14 @@
         object-fit: contain;
```
Comment (imnotjames) [inline frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss:274]: This has been tested mobile too, right?

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.ts [hunk 3]
```diff
@@ -758,41 +755,38 @@ export class CbxReaderComponent implements OnInit, OnDestroy {
   private updateFooterPage(): void {
     this.resetSwipeBoundaryHits();
     this.footerService.setCurrentPage(this.currentPage());
+    this.footerService.setTwoPageView(this.isTwoPageView() && this.getCurrentSpreadPages().length > 1);
```
Comment (imnotjames) [inline frontend/src/app/features/readers/cbx-reader/cbx-reader.component.ts]: Should this include the paginated mode check that single page layer has?

Should it be its own function?  It looks like it's used a few spots.

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.html [hunk 4]
```diff
@@ -67,13 +67,14 @@
     <div class="pages-wrapper" role="button" tabindex="0" (click)="onImageClick()" (keydown.enter)="onImageClick()"
       (dblclick)="onImageDoubleClick()">
       @if (isPageTransitioning() && previousImageUrls().length > 0) {
-      <div class="previous-page-layer">
+      <div class="previous-page-layer" [class.single-page-layer]="previousImageUrls().length === 1">
```
Comment (imnotjames) [inline frontend/src/app/features/readers/cbx-reader/cbx-reader.component.html]: Should this be a helper that includes the pagination check

File: frontend/src/app/features/readers/cbx-reader/core/cbx-spread.util.spec.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    width: widePages.includes(index) ? 1800 : 900,
+    height: 1200,
+    wide: widePages.includes(index)
+  }));
+}
+
+describe('cbx spread utilities', () => {
+  const pages = [1, 2, 3, 4, 5, 6];
+
+  it('keeps the cover solo and pairs even-starting spreads for LTR comics', () => {
```
Comment (imnotjames) [inline frontend/src/app/features/readers/cbx-reader/core/cbx-spread.util.spec.ts:25]: Should add a test case for empty

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss (hunk 1 above)
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss:367]: it's from the 2x 0.5 rems on the top/bottom

File: frontend/src/app/features/readers/cbx-reader/core/cbx-spread.util.spec.ts (hunk 5 above)
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/core/cbx-spread.util.spec.ts:25]: done

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.html (hunk 4 above)
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/cbx-reader.component.html]: done

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.ts (hunk 3 above)
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/cbx-reader.component.ts]: done

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss (hunk 2 above)
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss:274]: yep

**Review by imnotjames — APPROVED** [review]
Seems to work with testing and lgtm

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.ts [hunk 6]
```diff
@@ -758,41 +755,50 @@ export class CbxReaderComponent implements OnInit, OnDestroy {
...
+    return this.isPaginatedTwoPageView() && this.previousImageUrls().length === 1;
+  }
+
+  isSinglePageLayer(): boolean {
+    return this.isPaginatedTwoPageView() && !this.hasVisibleTwoPageSpread();
+  }
+
+  private getImageUrlsForCurrentPage(): string[] {
+    if (!this.pages().length) return [];
+    return this.getCurrentSpreadPages().map(pageIndex => this.getPageImageUrl(pageIndex));
```
[the remark this answers is not in the document]
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/cbx-reader.component.ts:797]: the backend dimension list should not be incomplete- if it is we've got worse problems on our hands

File: frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss [hunk 7]
```diff
@@ -369,9 +397,27 @@
...
+          }
+
+          .previous-page-layer,
+          .current-page-layer {
+            display: grid;
+            grid-template-columns: repeat(2, max-content);
+            width: max-content;
+            height: max-content;
+            justify-content: start;
+          }
```
[the remark this answers is not in the document]
Comment (zachyale) [reply frontend/src/app/features/readers/cbx-reader/cbx-reader.component.scss]: that's good- if it's larger than expected, it makes sense to start them out on the left, otherwise they might get spoiled if they see the center fold before they read the left side
